### PR TITLE
py-distance-class

### PR DESCRIPTION
**Pull request by muriakula**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1,3 +1,67 @@
...
+    def __init__(self, km: int) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Any) -> Any:
```
Comment (Bohdan-Salamakha) [inline app/main.py:14]: other can be int, float or Distance
return type is Distance

File: app/main.py [hunk 2]
```diff
@@ -1,3 +1,67 @@
...
+        if type(other) == int or type(other) == float:
+            new_distance = self.km + other
+            return Distance(new_distance)
+        else:
+            new_distance = self.km + other.km
+            return Distance(new_distance)
+
+    def __iadd__(self, other: Any) -> Any:
```
Comment (Bohdan-Salamakha) [inline app/main.py:22]: other can be int, float or Distance
return type is Distance

File: app/main.py [hunk 3]
```diff
@@ -1,3 +1,67 @@
...
+        if type(other) == int or type(other) == float:
+            self.km = self.km + other
+            return self
+        else:
+            self.km = self.km + other.km
+            return self
+
+    def __mul__(self, other: (int, float)) -> Any:
```
Comment (Bohdan-Salamakha) [inline app/main.py:30]: other can be int, float or Distance
return type is Distance

File: app/main.py [hunk 4]
```diff
@@ -1,3 +1,67 @@
...
+        if type(other) == int or type(other) == float:
+            new_distance = self.km * other
+            return Distance(new_distance)
+
+    def __truediv__(self, other: (int, float)) -> Any:
```
Comment (Bohdan-Salamakha) [inline app/main.py:35]: other can be int, float or Distance
return type is Distance

File: app/main.py [hunk 5]
```diff
@@ -1,3 +1,67 @@
...
+        if type(other) == int or type(other) == float:
+            new_distance = round(self.km / other, 2)
+            return Distance(new_distance)
+
+    def __eq__(self, other: Any) -> bool:
```
Comment (Bohdan-Salamakha) [inline app/main.py:40]: other can be int, float or Distance

File: app/main.py [hunk 6]
```diff
@@ -1,3 +1,67 @@
...
+        if type(other) == int or type(other) == float:
+            return self.km == other
+        return self.km == other.km
+
+    def __gt__(self, other: Any) -> bool:
```
Comment (Bohdan-Salamakha) [inline app/main.py:45]: other can be int, float or Distance

File: app/main.py [hunk 7]
```diff
@@ -1,3 +1,67 @@
...
+        if type(other) == int or type(other) == float:
+            return self.km > other
+        else:
+            return self.km > other.km
+
+    def __ge__(self, other: Any) -> bool:
```
Comment (Bohdan-Salamakha) [inline app/main.py:51]: other can be int, float or Distance

File: app/main.py [hunk 8]
```diff
@@ -1,3 +1,67 @@
...
+        if type(other) == int or type(other) == float:
+            return self.km >= other
+        else:
+            return self.km >= other.km
+
+    def __lt__(self, other: Any) -> bool:
```
Comment (Bohdan-Salamakha) [inline app/main.py:57]: other can be int, float or Distance

File: app/main.py [hunk 9]
```diff
@@ -1,3 +1,67 @@
...
+        if type(other) == int or type(other) == float:
+            return self.km < other
+        else:
+            return self.km < other.km
+
+    def __le__(self, other: Any) -> bool:
```
Comment (Bohdan-Salamakha) [inline app/main.py:63]: other can be int, float or Distance

File: app/main.py [hunk 10]
```diff
@@ -1,3 +1,67 @@
...
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Any) -> Any:
+        if type(other) == int or type(other) == float:
+            new_distance = self.km + other
+            return Distance(new_distance)
+        else:
```
Comment (Bohdan-Salamakha) [inline app/main.py:18]: remove redundant "else"

File: app/main.py [hunk 11]
```diff
@@ -1,3 +1,67 @@
...
+            new_distance = self.km + other.km
+            return Distance(new_distance)
+
+    def __iadd__(self, other: Any) -> Any:
+        if type(other) == int or type(other) == float:
+            self.km = self.km + other
+            return self
+        else:
```
Comment (Bohdan-Salamakha) [inline app/main.py:26]: remove redundant "else"

File: app/main.py [hunk 12]
```diff
@@ -1,3 +1,67 @@
...
+            self.km = self.km + other.km
+            return self
+
+    def __mul__(self, other: (int, float)) -> Any:
+        if type(other) == int or type(other) == float:
+            new_distance = self.km * other
+            return Distance(new_distance)
```
Comment (Bohdan-Salamakha) [inline app/main.py:33]: add condition for Distance

File: app/main.py [hunk 13]
```diff
@@ -1,3 +1,67 @@
...
+
+    def __eq__(self, other: Any) -> bool:
+        if type(other) == int or type(other) == float:
+            return self.km == other
+        return self.km == other.km
+
+    def __gt__(self, other: Any) -> bool:
+        if type(other) == int or type(other) == float:
+            return self.km > other
+        else:
```
Comment (Bohdan-Salamakha) [inline app/main.py:48]: remove redundant "else"

File: app/main.py [hunk 14]
```diff
@@ -1,3 +1,67 @@
...
+            return self.km > other.km
+
+    def __ge__(self, other: Any) -> bool:
+        if type(other) == int or type(other) == float:
+            return self.km >= other
+        else:
```
Comment (Bohdan-Salamakha) [inline app/main.py:54]: remove redundant "else"

File: app/main.py [hunk 15]
```diff
@@ -1,3 +1,67 @@
...
+            return self.km >= other.km
+
+    def __lt__(self, other: Any) -> bool:
+        if type(other) == int or type(other) == float:
+            return self.km < other
+        else:
```
Comment (Bohdan-Salamakha) [inline app/main.py:60]: remove redundant "else"

File: app/main.py [hunk 16]
```diff
@@ -1,3 +1,67 @@
...
+            return self.km < other.km
+
+    def __le__(self, other: Any) -> bool:
+        if type(other) == int or type(other) == float:
+            return self.km <= other
+        else:
```
Comment (Bohdan-Salamakha) [inline app/main.py:66]: remove redundant "else"

File: app/main.py [hunk 17]
```diff
@@ -1,3 +1,67 @@
...
+
+    def __mul__(self, other: (int, float)) -> Any:
+        if type(other) == int or type(other) == float:
+            new_distance = self.km * other
+            return Distance(new_distance)
+
+    def __truediv__(self, other: (int, float)) -> Any:
+        if type(other) == int or type(other) == float:
+            new_distance = round(self.km / other, 2)
+            return Distance(new_distance)
```
Comment (Bohdan-Salamakha) [inline app/main.py:38]: add condition for Distance

**Review by Bohdan-Salamakha — CHANGES_REQUESTED** [review]
v1

File: app/main.py [hunk 18]
```diff
@@ -1,3 +1,60 @@
...
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Union[int, float, "Distance"]) -> "Distance":
+        if type(other) == int or type(other) == float:
```
Comment (olena-hul) [inline app/main.py:15]: Use the opposite condition: `if isinstance(other, Distance)`
It's better to use `isinstance` than `type` for type checks

**Review by olena-hul — CHANGES_REQUESTED** [review]

**Pull request closed without merging**